### PR TITLE
feat(RHTAP-1933): add pipeline to sign binaries and release to GitHub

### PR DESCRIPTION
**Pull request by ergonzal22**
- Create new pipeline called released-to-github-signed, to make releases to GitHub with the checksum file signed
- Add base64-encode-checksum task that base64 encodes the checksum file to send it for signing
- WIP: Use create-internal-request task to send the encoded file to be signed
- Add collect-gh-params task to collect all the necessary GitHub information to send to the create-github-release task
- Update create-github-release task by deleting the installation of the gh cli, since it is incorporated in the release image

### CODE REVIEW

File: pipelines/release-to-github-signed/README.md
Comment (johnbieren) [inline pipelines/release-to-github-signed/README.md:3]: `s/signed a checksum/sign a checksum/`?

File: pipelines/release-to-github-signed/README.md
Comment (johnbieren) [inline pipelines/release-to-github-signed/README.md:15]: These three ec parameters were replaced with just the bundle. You probably copied a pipeline before the change, but can you switch it to be like the others now?

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (johnbieren) [inline pipelines/release-to-github-signed/release-to-github-signed.yaml:175]: I see you have the right parameter here, which is good (you can resolve this). This is the one that should be in the README

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (johnbieren) [inline pipelines/release-to-github-signed/release-to-github-signed.yaml:36]: Same comment as README

File: tasks/base64-encode-checksum/README.md
Comment (johnbieren) [inline tasks/base64-encode-checksum/README.md:9]: can you fix the formatting here? Looks like everything is aligned vertically except for the `|` after subdirectory

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (johnbieren) [inline tasks/base64-encode-checksum/base64-encode-checksum.yaml:15]: it's called image_binaries_path in the README. They should be in sync

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (johnbieren) [inline tasks/base64-encode-checksum/base64-encode-checksum.yaml:7]: should be 1.0.0

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (johnbieren) [inline tasks/base64-encode-checksum/base64-encode-checksum.yaml:33]: should use a redhat-appstudio image instead of hacbs-release

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (johnbieren) [inline tasks/base64-encode-checksum/base64-encode-checksum.yaml:36]: is this needed?

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (johnbieren) [inline tasks/base64-encode-checksum/base64-encode-checksum.yaml:29]: If it is a file, shouldn't it be in a workspace so downstream things can use it?

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (johnbieren) [inline tasks/collect-gh-params/collect-gh-params.yaml:7]: should be 1.0.0

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (johnbieren) [inline tasks/collect-gh-params/collect-gh-params.yaml:43]: should be redhat-appstudio image

**Review by johnbieren — COMMENTED** [review]
Some comments as part of an initial pass through as it is still a draft

File: pipelines/release-to-github-signed/README.md
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/README.md]: Changed

File: pipelines/release-to-github-signed/README.md
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/README.md]: Replaced! Thank you!

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/release-to-github-signed.yaml]: Done!

File: tasks/base64-encode-checksum/README.md
Comment (ernesgonzalez33) [reply tasks/base64-encode-checksum/README.md]: Fixed

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (ernesgonzalez33) [reply tasks/base64-encode-checksum/base64-encode-checksum.yaml]: Done!

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (ernesgonzalez33) [reply tasks/base64-encode-checksum/base64-encode-checksum.yaml]: No, it's not! Deleted!

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (ernesgonzalez33) [reply tasks/base64-encode-checksum/base64-encode-checksum.yaml]: I changed the description a bit. It's not a file, but the output of the file after being base64 encoded

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (ernesgonzalez33) [reply tasks/collect-gh-params/collect-gh-params.yaml]: Changed

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (ernesgonzalez33) [reply tasks/collect-gh-params/collect-gh-params.yaml]: Changed here and in every task modified in this PR

File: tasks/create-github-release/create-github-release.yaml
Comment (johnbieren) [inline tasks/create-github-release/create-github-release.yaml:50]: should be `REPOSITORY` not `REPO` at the end of the line but it doesn't really matter much since it is commented out

File: tasks/create-github-release/tests/test-create-github-release.yaml
Comment (johnbieren) [inline tasks/create-github-release/tests/test-create-github-release.yaml:75]: Why are we removing the check that the url result is what we expect?

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (johnbieren) [inline tasks/collect-gh-params/collect-gh-params.yaml:63]: Note to release-team, if we complete https://issues.redhat.com/browse/RHTAPREL-668, we don't need checks like this. Linking this line to that ticket

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (johnbieren) [inline pipelines/release-to-github-signed/release-to-github-signed.yaml:47]: Should these be removed or what is the plan for them? If you are waiting to turn them on later once another task is added to the pipeline, can you add a comment saying so?

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (johnbieren) [inline pipelines/release-to-github-signed/release-to-github-signed.yaml:140]: I don't have the background context for this pipeline, but are we expecting it to only run for applications with a single component? Want to make sure the inclusion of this task wasn't just copied over and forgotten about

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/release-to-github-signed.yaml]: Thank you for the heads-up! I removed it

File: tasks/create-github-release/create-github-release.yaml
Comment (ernesgonzalez33) [reply tasks/create-github-release/create-github-release.yaml]: Good catch! Fixed

File: tasks/create-github-release/tests/test-create-github-release.yaml
Comment (ernesgonzalez33) [reply tasks/create-github-release/tests/test-create-github-release.yaml:75]: Because we are not using the  `echo` of the release url at the end of the task anymore to send to the results. I found it cleaner to pipe the output of the `gh` command since it outputs the URL that we need. And since in the tests what we are doing is mocking that `gh`  command, I don't see the point of testing an  `echo` that I manually need to create in the `mock.sh`

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/release-to-github-signed.yaml:47]: Updated the comment

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/release-to-github-signed.yaml:140]: For the terraform pipeline, we are only covering one component to sign and release to github. The rest of the pipeline is not compatible to releasing more than one components at the moment, so I kept this test because of that

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (mmalina) [inline pipelines/release-to-github-signed/release-to-github-signed.yaml:15]: Yesterday, we merged this PR that added git resolver parameters to each pipeline: https://github.com/redhat-appstudio/release-service-catalog/pull/298

So these parameters should be added here (along with using them in the task references).

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (ernesgonzalez33) [reply tasks/base64-encode-checksum/base64-encode-checksum.yaml]: But that would mean giving the `extract-binaries-from-image` task two different functionalities. Maybe we need in  the future to extract some binaries but not base64 encode anything. Or maybe we would need to base64 encode something without extracting anything and we can only make this task more generic without modifying the other one

File: pipelines/release-to-github-signed/README.md
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/README.md:1]: That's fine by me! I'll change it

File: pipelines/release-to-github-signed/release-to-github-signed.yaml
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/release-to-github-signed.yaml]: Updated!

File: pipelines/release-to-github-signed/README.md
Comment (ernesgonzalez33) [reply pipelines/release-to-github-signed/README.md:1]: Changed!

File: tasks/create-github-release/create-github-release.yaml
Comment (ernesgonzalez33) [reply tasks/create-github-release/create-github-release.yaml]: You are right! Removed

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (mmalina) [reply tasks/base64-encode-checksum/base64-encode-checksum.yaml]: I took a look at the changes again to understand why we even need to encode it. And I'm not sure. This will be passed to the `sign-checksum-file` which doesn't exist yet. But why does it need the file to be base64 encoded? Won't the task decode it anyway? I'd expect that what we'll want to sign is probably a sha sum of the checksum file, no?

If the `sign-checksum-file` task needs the original checksum file, why not pass it via workspace? That's a better fit for files usually - we switched to that in most places (e.g. collect-data). The main concern when passing data via results is its size - there is a 4k size limit. But if that's not a concern, we could just pass the whole file's contents directly - why bother with base64? If anything, it will make it even bigger.

File: pipelines/release-to-github-signed/README.md
Comment (mmalina) [reply pipelines/release-to-github-signed/README.md:1]: Cool, thanks.

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (ernesgonzalez33) [reply tasks/collect-gh-params/collect-gh-params.yaml:1]: Added: https://github.com/redhat-appstudio/release-service-catalog/pull/281/files#diff-3d59b1a68aab45183ec7a518930334ffa07207906e39c2a534a24d57cad49550

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (ernesgonzalez33) [reply tasks/base64-encode-checksum/base64-encode-checksum.yaml]: I need to base64 encode the checksum file to pass it as parameter to the internal request, which is going to trigger the internal pipeline and send that blob to sign it. That's why I created this task. All the signing stuff is still WIP so maybe this task could change. If you prefer, I can comment it in the meantime that's finished

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (johnbieren) [reply tasks/base64-encode-checksum/base64-encode-checksum.yaml:15]: @ernesgonzalez33

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (johnbieren) [inline tasks/collect-gh-params/collect-gh-params.yaml:64]: this echo seems wrong

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (mmalina) [inline tasks/collect-gh-params/collect-gh-params.yaml:61]: It took me a bit to realize what's stored in this variable. What do you think about changing the variable name to something like `CHECKSUMS_FILE`? I think it would make it more obvious what it's for.

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (mmalina) [reply tasks/collect-gh-params/collect-gh-params.yaml:1]: Cool

File: tasks/base64-encode-checksum/base64-encode-checksum.yaml
Comment (mmalina) [reply tasks/base64-encode-checksum/base64-encode-checksum.yaml]: OK, thanks. I still think this could be easily done in the signing task, but I guess it's fine to do it separately as well.

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (ernesgonzalez33) [reply tasks/collect-gh-params/collect-gh-params.yaml]: Because it is wrong! Changed!

File: tasks/collect-gh-params/collect-gh-params.yaml
Comment (ernesgonzalez33) [reply tasks/collect-gh-params/collect-gh-params.yaml]: Makes sense! Changed

**Review by mmalina — DISMISSED** [review]

**Review by johnbieren — DISMISSED** [review]

**Comment by ernesgonzalez33** [comment]
Now it can be reviewed again. I forgot to change the `taskGitUrl` and `taskGitRevision` to `redhat-appstudio` and `main`

File: pipelines/release-to-github/release-to-github.yaml
Comment (theflockers) [inline pipelines/release-to-github/release-to-github.yaml:53]: Don't forget to remove this before merging. - edit - or do like I said in the next comment.

File: pipelines/release-to-github/release-to-github.yaml
Comment (theflockers) [inline pipelines/release-to-github/release-to-github.yaml:259]: If this block depends on a internal Pipeline you can either wait until that pipeline is merged to move forward or remove this block and add it back in a next PR. We don't keep such big comment blocks in our pipelines/tasks.

**Review by theflockers — CHANGES_REQUESTED** [review]

File: pipelines/release-to-github/release-to-github.yaml
Comment (ernesgonzalez33) [reply pipelines/release-to-github/release-to-github.yaml]: Deleted. I have it in another branch

File: pipelines/release-to-github/release-to-github.yaml
Comment (ernesgonzalez33) [reply pipelines/release-to-github/release-to-github.yaml]: Deleted

**Review by theflockers — APPROVED** [review]

**Pull request merged**
